### ISSUE TITLE
Producer timeout when update to v0.4.9

### ISSUE THREAD

**Issue by arthurhenrique** [segmentio/kafka-go#594]
**Describe the bug**
Producer timeout when update to v0.4.9

**Kafka Version**
confluentinc/cp-kafka:4.1.3

**To Reproduce**
```golang
err := s.producer.WriteMessages(s.ctx, kafka.Message{
			Key:   []byte(Id),
			Topic: Xptopic,
			Value: data,
		})
```
**Expected behavior**
TImeout

**Additional context**

**Comment by achille-roussel** [comment]
Hello @arthurhenrique, thanks for reporting this issue.

Would you be able to share which version of kafka-go you were using prior to v0.4.9?

**Comment by arthurhenrique** [comment]
Hey... Of course it's v0.4.8 :+1:

**Comment by arthurhenrique** [comment]
Hello @achille-roussel,

I've had to debug this problem and I realized that Topic definition now is mutually exclusive on version v0.4.9, which returns a [CORRUPT_MESSAGE](https://kafka.apache.org/protocol#protocol_error_codes) error message when the topic name is defined on Context and Message structs.

refs: https://github.com/segmentio/kafka-go/pull/438/files#diff-6326c99dc4638d1cf8582c8ab1941867b252a173a640030c64678793f10a3b5cR979

Just for curiosity, why this has changed?

Thank you

**Comment by achille-roussel** [comment]
I'm glad you got to the bottom of your original issue.

As to why we changed the behavior; with version 0.4, we introduced the ability for the `kafka.Writer` to produce to multiple topics. It now has two modes, either the topic is defined on the writer (and the writer produces to a single topic, in which case the messages must not set a topic), or there is not topic configured on the writer and the messages are expected to contain the topic that they must be delivered to.

We made the decision to explicitly refuse messages that set a topic when they are written to a writer that has a topic configure to avoid implicitly ignoring the topic set on messages. Obviously it's a trade off, and we're open to hearing feedback and change the behavior if we didn't make the right one.
